content/doc/book/managing/plugins.adoc

```diff
@@ -4,13 +4,257 @@ layout: section
+
+==== On the master
+
+Assuming a `.hpi` file has been explicitly downloaded by a systems
+administrator, the administrator can manually place the `.hpi` file in a
+specific location on the file system.
+
+Copy the downloaded `.hpi`` file into the `JENKINS_HOME/plugins` directory on
+the Jenkins master (for example, on Debian systems `JENKINS_HOME` is generally
+`/var/lib/jenkins`).
+
+The master will need to be restarted before the plugin is loaded and
+made available in the Jenkins environment.
+
 [NOTE]
 ====
-This is still very much a work in progress
+The names of the plugin directories in the Update Site footnoteref:[uc] are
+not always the same as the plugin's display name. Searching
+link:https://plugins.jenkins.io/[plugins.jenkins.io]
+for the desired plugin will provide the appropriate link to the `.hpi` files.
+====
+
+== Updating a plugin
+
+Updates are listed in the *Updates* tab of the *Manage Plugins* page and can be
+installed by checking the checkboxes of the desired plugin updates and clicking
+the *Download now and install after restart* button.
+
+image::managing/plugin-manager-update.png["Updates tab in the Plugin Manager", role=center]
+By default, the Jenkins master will check for updates from the Update Center
+once every 24 hours. To manually trigger a check for updates, simply click on
+the *Check now* button in the *Updates* tab.
+
+== Removing a plugin
+
+When a plugin is no longer used in a Jenkins environment, it is prudent to
+remove the plugin from the Jenkins master. This provides a number of benefits
+such as reducing memory overhead at boot or runtime, reducing configuration
+options in the web UI, and removing the potential for future conflicts with new
+plugin updates.
+
+=== Uninstalling a plugin
+
+The simplest way to uninstall a plugin is to navigate to the *Installed* tab on
+the *Manage Plugins* page. From there, Jenkins will automatically determine
+which plugins are safe to uninstall, those which are not dependencies of other
+plugins, and present a button for doing so.
+
+image::managing/plugin-manager-uninstall.png["Installed tab in the Plugin Manager", role=center]
+
+
+A plugin may also be uninstalled by removing the corresponding `.hpi`
+file from the `JENKINS_HOME/plugins` directory on the master. The plugin will
+continue to function until the master has been restarted.
+
+[CAUTION]
+====
+If a plugin `.hpi` file is removed but required by other plugins, the Jenkins
+master may fail to boot correctly.
+====
+
+Uninstalling a plugin does *not* remove the configuration that the plugin may
+have created. If there are existing jobs/nodes/views/builds/etc configurations
+that reference data created by the plugin, during boot Jenkins will warn that
+some configurations could not be fully loaded and ignore the unrecognized data.
+
+Since the configuration(s) will be preserved until they are overwritten,
+re-installing the plugin will result in those configuration values reappearing..
```

Comment:
You might also spell out that this means that if you reinstall the plugin its full configuration is restored.